File: app/src/main/java/com/softaai/designpatterns/java/structural/facade/FacadeDesignPatternDemo.java
package com.softaai.designpatterns.java.structural.facade;

/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
public class FacadeDesignPatternDemo {
    public static void main(String[] args){
        ShapeMaker shapeMaker = new ShapeMaker();
        shapeMaker.drawRectangle();
        shapeMaker.drawSquare();
        shapeMaker.drawCircle();
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/Facade/FacadeDesignPatternDemo.kt
package com.softaai.designpatterns.kotlin.structural.Facade


/**
 * Created by Amol Pawar on 03-04-2019.
 * softAai Apps
 */
object FacadeDesignPatternDemo {
    @JvmStatic
    fun main(args : Array<String>){
        val shapeMaker = ShapeMaker()
        shapeMaker.drawRectangle()
        shapeMaker.drawSquare()
        shapeMaker.drawCircle()
    }
}